arch/posix/include/posix_cheats.h

```diff
@@ -12,8 +12,6 @@
 #if !defined(ZEPHYR_ARCH_POSIX_INCLUDE_POSIX_CHEATS_H_) && !defined(NO_POSIX_CHEATS)
 #define ZEPHYR_ARCH_POSIX_INCLUDE_POSIX_CHEATS_H_
 
-#ifdef CONFIG_ARCH_POSIX
```

Comment:
Well, I'm not sure about removal of this guard. I'd rather have it stay, or worst-case, we'll have behavior changes outside ARCH_POSIX, whereas the aim to get it right for ARCH_POSIX.

Reply:
> Well, I'm not sure about removal of this guard.

Ok, now I see that comments say that this header is included only for ARCH_POSIX anyway.